grid-proxy/internal/gpuindexer/gpuindexer.go

```diff
@@ -60,13 +68,67 @@ func NewNodeGPUIndexer(
 	return indexer, nil
 }
 
+// startDBListener sets up a PostgreSQL listener to listen for changes in the database and triggers the nodesChangeChan channel.
+func (n *NodeGPUIndexer) startDBListener(ctx context.Context, psqlInfo string) {
+	listener := pq.NewListener(psqlInfo, minListenerReconnectInterval, 6*minListenerReconnectInterval, func(ev pq.ListenerEventType, err error) {
+		if err != nil {
+			log.Error().Err(err).Msg("failed listening to DB changes")
+		}
+	})
+	defer listener.Close()
+
+	err := listener.Listen(dbNotificationChannel)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to listen to DB changes")
+		return
```

Comment:
if this ever happens, this will fail and will never recover, that startDBListener should always be running, if you exit this routine silently, what do we do? we fall back completely to the 1hr scheduler? is that the intended behavior?

Reply:
yes this exactly what will happen